lib/engine/train.rb

```diff
@@ -28,7 +28,7 @@ def initialize(name:, distance:, price:, index: 0, **opts)
       @rusted = false
       @obsolete = false
       @operated = false
-      @events = (opts[:events] || []).select { |e| @index == (e[:when] || 0) }
+      @events = (opts[:events] || []).select { |e| @index == (e[:when] || 1) - 1 }
```

Comment:
Event definitions shouldn't need to know train indexes are zero based. For example, if :when = 3, it is expected that the event happens on the 3rd train, not the 4th.